src/Metadata/Extractor/ResourceExtractorTrait.php

```diff
@@ -97,21 +97,39 @@ private function buildArgs(SimpleXMLElement $resource): ?array
         return $data;
     }
 
-    /**
-     * @return string[]
-     */
     private function buildValues(SimpleXMLElement $resource): array
     {
         $data = [];
         foreach ($resource->value as $value) {
             if (null !== $value->attributes()->name) {
-                $data[(string) $value->attributes()->name] = isset($value->values) ? $this->buildValues($value->values) : (string) $value;
+                $data[(string) $value->attributes()->name] = isset($value->values) ? $this->buildValues($value->values) : $this->castValue($value->__toString());
                 continue;
             }
 
-            $data[] = isset($value->values) ? $this->buildValues($value->values) : (string) $value;
+            $data[] = isset($value->values) ? $this->buildValues($value->values) : $this->castValue($value->__toString());
         }
 
         return $data;
     }
+
+    private function castValue($value)
```

Comment:
Could we use something from Symfony which already does it?

Reply:
Good point! [`Symfony\Component\Config\Util\XmlUtils::phpize`](https://github.com/symfony/config/blob/6.1/Util/XmlUtils.php#L210) seems to do the job, except for `'1'`, `1`, `'0'` and `0` values